packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -178,6 +178,18 @@ class RequestWrapper {
   async fetch({request}) {
     assert.atLeastOne({request});
 
+    if (this.behaviorCallbacks.requestWillFetch) {
+      for (let callback of this.behaviorCallbacks.requestWillFetch) {
+        const returnedRequest = callback({request});
+        if (returnedRequest && returnedRequest.then) {
```

Comment:
+1 Jeff's comment and if we role with this please change the check for an explicit Promise rather than a check for the "then" parameter